plugins/kapt3/kapt3-base/src/org/jetbrains/kotlin/kapt3/base/incremental/javacVisitors.kt

```diff
@@ -196,12 +198,23 @@ private class TypeTreeVisitor(val elementUtils: Elements, val trees: Trees, val
     }
 }
 
+private val literalKinds = setOf(
+    Tree.Kind.BOOLEAN_LITERAL,
+    Tree.Kind.INT_LITERAL,
+    Tree.Kind.LONG_LITERAL,
+    Tree.Kind.DOUBLE_LITERAL,
+    Tree.Kind.FLOAT_LITERAL,
+    Tree.Kind.CHAR_LITERAL,
+    Tree.Kind.STRING_LITERAL
+)
+
 /**
  * Visits a constant initializer expression, and extracts all references to constants, either through field select (A.MY_FIELD) or
  * identifier (MY_FIELD, which happens if A.MY_FIELD is statically imported).
  */
 private class ConstantTreeVisitor(val sourceStructure: SourceFileStructure) : TreePathScanner<Void, Void>() {
 
+    fun ifTreeIsLiteral(tree: Tree) = tree.kind !in literalKinds
```

Comment:
Either the name or the function body looks counterintuitive to me :)
Maybe rename it to `isNonTrivialLiteralExpression()` or something similar?

Reply:
Thanks for spotting this. I've renamed it to `isTrivialLiteralExpression`, to make it more readable.